### PR TITLE
Intensity graph with log scales

### PR DESCRIPTION
**Pull request by cristiingineru**
# Tests reorg
* The tests for color scale were moved in their own file
* The helper functions used to generate test matrixes and get the pixel color were moved in a different file test/utils/colorMatchers.js
* Adding a Jasmine matcher in test/utiles/colorMatchers.js for testing a pixel color. This matcher take into account the pixel ratio which is important because the tests that are using it are passing on any screen regardless the DPI (on my laptop lots of tests of the intensity graph are failing because they assume the pixel ratio = 1). Eventually all tests should be changed to use this matcher.

# Support for log scales (axes)
* Until now only the linear scaling was supported. In order to optimize one of the 2 modes were used based on data points per pixel density: rect by rect (when the data is small) and pixel by pixel (when the data matrix is huge and it has to be decimated)
* The new mode can handle any type of mapping - log or anything else as long as the p2c and c2p functions of the axis can be used to determine how many pixels and where each data point needs to be drown
* The new mode is very very slow - up to a few frames per seconds regardless the data size (!)
* If the time allows then some caching about the size and position of the data will be made.

### CODE REVIEW

File: tests/utils/colorMatchers.js [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+                    cy = (plotOffset.top + y) * pixelRatio,
+                    actual = getPixelColor(ctx, cx, cy),
+                    result = {};
+                result.pass = isClose(actual, expected);
+                if (!result.pass) {
+                    result.message =
+                      'Expected ' + printColor(expected) +
+                      ' at ' + x + ',' + y +
+                      ' / ' + cx + ',' + cy +
+                      ' actual ' + printColor(actual);
```
Comment (cristiingineru) [inline tests/utils/colorMatchers.js]: The detailed error message was needed to understand why the tests were failing on phantomjs.

File: jquery.flot.intensitygraph.js [hunk 2]
```diff
@@ -232,19 +283,51 @@ function IntensityGraph() {
...
+                            if (value2 > value) {
+                                value = value2;
+                            }
+                        }
+                        for (y2 = y; y2 < y2limit; y2++) {
+                            value2 = data[x][y2];
+                            if (value2 > value) {
+                                value = value2;
+                            }
+                        }
```
Comment (cristiingineru) [inline jquery.flot.intensitygraph.js]: The last 2 for statements are used to cover some corner cases when the column or row that have to be decimated (looking for the highest data point) is 1 point wide only. The time spent by them is minimal compared with the p2c and c2p calls.

**Comment by cipix2000** [comment]
maybe you can one more test to get the coverage back to 100 %

**Comment by cristiingineru** [comment]
I've added one example in the webcharts-drive-test and modified the performance example to switch between linear and log mode. The results are terrible: up to 5 fps regardless how small the data matrix is. The bigger the drawing area the slower the log.

Do you think it's ok to add to this repo an example which depends on the node_module/engineering-flot/jquery.flot.logaxis.js? In theory if that plugin is missing then the axes should default to linear.

**Comment by cristiingineru** [comment]
Definitely a few tests to get back to 100% coverage.

**Pull request merged**